contracts/src/LiquidityPoolV2.sol

```diff
@@ -50,14 +79,15 @@ contract LiquidityPoolV2 is ReentrancyGuardUpgradeable, AccessManagedUpgradeable
         // Calculate liquidity tokens to mint
         uint256 liquidity;
         if (totalLiquidity == 0) {
+            // For the first liquidity provider, mint liquidity tokens 1:1 with the provided tokens
             liquidity = Math.sqrt(tokenAAmount * tokenBAmount);
```

Comment:
**adjustForDecimals** (see last comment of the file to understand)

Replace tokenAAmount and tokenBAmount in the transfers but the adjustedToken